internal/controller/sleepinfo/jsonpatch/genericresources.go

```diff
@@ -72,18 +72,36 @@ func (g genericResource) getListOptions(namespace string, target v1alpha1.PatchT
 		Limit:     500,
 	}
 
+	includeRef := g.ResourceClient.SleepInfo.GetIncludeRef()
 	excludeRef := g.ResourceClient.SleepInfo.GetExcludeRef()
+	fieldsToInclude := getFieldToInclude(includeRef, target)
+	labelsToInclude := getLabelsToInclude(includeRef)
 	fieldsToExclude := getFieldToExclude(excludeRef, target)
 	labelsToExclude := getLabelsToExclude(excludeRef)
-	if len(fieldsToExclude) > 0 {
+	if len(fieldsToInclude) > 0 {
+		fieldSelector, err := fields.ParseSelector(strings.Join(fieldsToInclude, ","))
+		if err != nil {
+			return nil, err
+		}
+		listOptions.FieldSelector = fieldSelector
+	}
+
+	if len(labelsToInclude) > 0 {
+		labelSelector, err := labels.Parse(strings.Join(labelsToInclude, ","))
+		if err != nil {
+			return nil, err
+		}
+		listOptions.LabelSelector = labelSelector
+	}
+	if len(fieldsToExclude) > 0 && len(fieldsToInclude) == 0 {
```

Comment:
So, if include something it is not possible to exclude?
I think about something like:
in a namespace, I have a set of deployment of the project 1, which have a `project: '1'` label. In this way, it would be possible to include only the deployments with this label, but if we also need to exclude a specific service?
What do you think?

Reply:
I assumed if a person wants to use includes, he/she won't be using excludes. But you're right about the use case you mentioned. I'll work on making that happen but I also need to validate a case where a person sets the same rule on both include and exclude fields since that's an invalid case if I'm right.

Reply:
@davidebianchi any opinions about my previous reply?

Reply:
I think we could rely on K8s: if the user set include and exclude which overlap, k8s will return empty value. 

We could also set a validation to give a warning to the user, but there could be also some complex cases to handle. If a kind is excluded by labels and included by name, and the resource with the name also has the resource what happens? Give a warning on this type of condition is quite complicated.

Reply:
Thanks for your reply! I agree with you on complexity of warning on client side. It can make the implementation quite complex. 

I'll update this PR to allow both inclusion and exclusions at the same sleep info resource and rely on k8s for the validation of having same rules on both exclusion and inclusion.